Add abilitiy to take screenshots with a key shortcut

dmarcos opened the pull request:
It introduces a scene level component `screenshot` that allows to take a snapshot of the rendered scene with the `alt-S` shortcut. It is configurable to take either equirectangular (panoramas) or screen snapshots. It should also work with the inspector. @chriscar I believe you were asking for this a couple of days ago in Slack. It would be great to get your feedback on it.

I still have to organized the code a little bit better but the functionality is complete. 

- Bumps three.js r81

chriscar commented:
Thanks for letting me know, will have a look. But I have to create a screenshot programmatically, not manually.

AdaRoseEdwards commented:
Something like `document.querySelector('[screenshot]').components.screenshot.capture()` should work @chriscar

ngokevin commented on tests/shaders/standard.test.js:
```diff
@@ -26,7 +26,6 @@ suite('standard material', function () {
     el.setAttribute('material', 'ambientOcclusionMapIntensity: 0.4; ambientOcclusionMap: url(' + imageUrl + ');');
     assert.equal(el.getObject3D('mesh').material.aoMapIntensity, 0.4);
     el.addEventListener('materialtextureloaded', function (evt) {
-      assert.equal(evt.detail.texture.image.getAttribute('src'), imageUrl);
```
Curious what three.js changed here?

ngokevin commented on tests/components/scene/screenshot.test.js:
```diff
@@ -0,0 +1,21 @@
+    document.body.appendChild(el);
+  });
+
+  test('capture is called when key shortut is pressed', function () {
```
typo shortcut

ngokevin commented on docs/components/screenshot.md:
```diff
@@ -0,0 +1,26 @@
+---
+
+Take screenshots of the scene using a keboard shortcut (alt+s).
+It can be configured to either take equirectangular captures (panoramas) or
```
```
360&deg; captures (`equirectangular`) or regular screenshots (`projection`)
```

ngokevin commented on docs/components/screenshot.md:
```diff
@@ -0,0 +1,26 @@
+It can be configured to either take equirectangular captures (panoramas) or
+regular screenshots. The screenshot component only applies to the [`<a-scene>` element][scene].
+
+## Example
```
I'd mention that it's a default component

ngokevin commented on docs/components/screenshot.md:
```diff
@@ -0,0 +1,26 @@
+|--------------|------------------------------------------------------------|---------------|
+| width        | The width in pixels of the screnshot taken.                | 4096          |
+| height       | The height in pixels of the screenshot taken.              | 2048          |
+| projection   | The screenshot projection: perspective or equirectangular | perspective   |
```
Should `equirectangular` be default? Seems more useful / cooler.

ngokevin commented on src/components/scene/screenshot.js:
```diff
@@ -0,0 +1,206 @@
+/* global ImageData, URL */
+var register = require('../../core/component').registerComponent;
```
registerComponent

ngokevin commented:
The perspective screenshot is upside down and reverse

dmarcos commented:
I made a bunch of changes based on review and fixed the vertically flipped pixels on equirectangular projection.

ngokevin commented:
There are still a few typo/doc comments to address in the diff.

Also, curious what changed about how three.js stores `<img>`s? Otherwise, r+wc

dmarcos commented:
I fixed the typos and changed the shortcut to `alt+ctrl+s`. `alt+s` opens the history menu on Firefox. Using `alt+ctrl+s` also makes it consistent with the inspector shortcut
